torch/distributed/_tensor/dispatch.py

```diff
@@ -125,6 +125,12 @@ def _operator_dispatch(
     kwargs: Dict[str, object],
     sharding_propagator: ShardingPropagator,
 ) -> Tuple[object, OpSchema, OutputSharding]:
+    # operators that does not need to go through sharding propagation
+    if op_call == aten.is_same_size.default:
```

Comment:
maybe not in this PR. But down the road, IMHO, we want to avoid this type check(for single or certain ops) as much as possible?

Reply:
I refactored the dispatch code a bit to not have the op type checks, now all things are handled by the `custom_op_handlers`